src/routes/run_group.rs

```diff
@@ -0,0 +1,518 @@
+    pool: web::Data<db::DbPool>,
+) -> HttpResponse {
+    // Query DB for results in new thread
+    match web::block(move || {
+        let conn = pool.get().expect("Failed to get DB connection from pool");
+
+        match RunGroupWithGithubData::find(&conn, query) {
+            Ok(val) => Ok(val),
+            Err(e) => {
+                error!("{}", e);
+                Err(e)
+            }
+        }
+    })
+    .await
+    {
+        Ok(results) => {
+            // If there are no results, return a 404
+            if results.is_empty() {
+                HttpResponse::NotFound().json(ErrorBody {
+                    title: "No run group found".to_string(),
+                    status: 404,
+                    detail: "No run group found with the specified parameters".to_string(),
+                })
+            } else {
+                // If there is no error, return a response with the retrieved data
+                HttpResponse::Ok().json(results)
+            }
+        }
+        Err(e) => {
+            error!("{}", e);
+            // If there is an error, return a 500
+            default_500(&e)
+        }
+    }
+}
+
+/// Handles DELETE requests to /run-groups/{id} for deleting run_group rows by run_group_id
+///
+/// This function is called by Actix-Web when a delete request is made to the /run-groups/{id}
+/// mapping
+/// It parses the id from `req`, connects to the db via a connection from `pool`, and attempts to
+/// delete the specified run group, returning the number or rows deleted or an error message if some
+/// error occurs
+///
+/// # Panics
+/// Panics if attempting to connect to the database results in an error
+async fn delete_by_id(req: HttpRequest, pool: web::Data<db::DbPool>) -> HttpResponse {
+    // Pull id param from path
+    let id = &req.match_info().get("id").unwrap();
+
+    // Parse ID into Uuid
+    let id = match parse_id(id) {
+        Ok(id) => id,
+        Err(error_response) => return error_response,
+    };
+
+    //Query DB for pipeline in new thread
+    match web::block(move || {
+        let conn = pool.get().expect("Failed to get DB connection from pool");
+        // Delete run_group_is_from_github record if there is one
+        if let Err(e) = RunGroupIsFromGithubData::delete_by_run_group_id(&conn, id) {
+            error!("{}", e);
+            return Err(e);
+        }
+
+        match RunGroupData::delete(&conn, id) {
+            Ok(delete_count) => Ok(delete_count),
+            Err(e) => {
+                error!("{}", e);
+                Err(e)
+            }
+        }
+    })
+    .await
+    {
+        // If there is no error, verify that a row was deleted
+        Ok(results) => {
+            if results > 0 {
+                let message = format!("Successfully deleted {} row", results);
+                HttpResponse::Ok().json(json!({ "message": message }))
+            } else {
+                HttpResponse::NotFound().json(ErrorBody {
```

Comment:
I still think it's worth trying to consolidate some of these similar calls into helper methods, but this can be done later as a refactoring process.

Reply:
Yeah, I would really love to do that if I ever have time

Reply:
👍 

This is another task that someone who is ramping up on the codebase can do.